### PR TITLE
fix(master): re-fix mismatching between information of master and datanode bug …

### PR DESCRIPTION
**Pull request by zhang111feng**
…when adding replica

**What this PR does / why we need it**:
re-fix the bug of problem mentioned in the previous pr.

**Which issue this PR fixes**:

re-fixes pr https://github.com/cubefs/cubefs/pull/2920

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by Victor1319 — DISMISSED** [review]
LGTM

File: master/cluster.go [hunk 1]
```diff
@@ -2576,9 +2574,9 @@ func (c *Cluster) addDataPartitionRaftMember(dp *DataPartition, addPeer proto.Pe
 	}
 
 	log.LogInfof("action[addDataPartitionRaftMember] try host [%v] to [%v] peers [%v] to [%v]",
-		dp.Hosts, newHosts, dp.Peers, newPeers)
+		dp.Hosts, dp.Hosts, dp.Peers, dp.Peers)
 
-	if err = dp.update("addDataPartitionRaftMember", dp.VolName, newPeers, newHosts, c); err != nil {
+	if err = dp.update("addDataPartitionRaftMember", dp.VolName, dp.Peers, dp.Hosts, c); err != nil {
```
Comment (sejust) [inline master/cluster.go:2579]: `MUST` restore hosts and peers when updating failed.

**Review by sejust — CHANGES_REQUESTED** [review]

File: master/cluster.go (hunk 1 above)
Comment (zhang111feng) [reply master/cluster.go:2579]: Thanks for your correction, I have modified this problem as you said.

Comment (zhang111feng) [reply master/cluster.go:2579]: > `MUST` restore hosts and peers when updating failed.

Thanks for your correction, I have modified this problem as you said.

**Review by sejust — APPROVED** [review]

**Review by sejust — APPROVED** [review]

**Review by Victor1319 — APPROVED** [review]
LGTM

**Pull request merged**